In [49]:
import pandas as pd

# Importing WindSpeed and WindDirection data 

In [50]:
WindSpeed = pd.read_csv("C:\Users\jonat\OneDrive\Documents\Python\Data\EDIT\FINO\FINO1_WindSpeed_90m_20040101_20151231", header = 0, delimiter= '\t', comment='#')
#WindSpeed['Time'] = pd.to_datetime(WindSpeed['Time'], format = '%Y-%m-%d %H:%M:%S')
WindSpeed.describe()

Value        Minimum        Maximum      Deviation   Quality
count  631149.000000  631149.000000  631149.000000  631149.000000  631149.0
mean      -45.388633     -46.975996     -43.535599     -64.008364       2.0
std       228.582874     227.747248     228.620361     245.907712       0.0
min      -999.000000    -999.000000   -1000.000000    -999.000000       2.0
25%         5.310000       3.770000       6.590000       0.300000       2.0
50%         8.550000       6.690000      10.180000       0.530000       2.0
75%        12.130000      10.050000      14.100000       0.850000       2.0
max        37.610000    1000.000000     999.000000       7.860000       2.0

In [51]:
WindDirection = pd.read_csv("C:\Users\jonat\OneDrive\Documents\Python\Data\EDIT\FINO\FINO1_WindDirection_90m_20040101_20151231", header = 0, delimiter= '\t', comment='#')
#WindDirection['Time'] = pd.to_datetime(WindDirection['Time'], format = '%Y-%m-%d %H:%M:%S')
WindDirection.describe()

Value       Minimum       Maximum      Deviation   Quality
count  631144.000000  6.311440e+05  6.311440e+05  631144.000000  631144.0
mean      127.933512 -9.999900e+02 -9.999900e+02     -56.730087       2.0
std       293.455589  7.010732e-09  7.010732e-09     246.932349       0.0
min      -999.000000 -9.999900e+02 -9.999900e+02    -999.000000       2.0
25%       101.700000 -9.999900e+02 -9.999900e+02       2.420000       2.0
50%       204.700000 -9.999900e+02 -9.999900e+02       3.250000       2.0
75%       264.400000 -9.999900e+02 -9.999900e+02       4.670000       2.0
max       360.000000 -9.999900e+02 -9.999900e+02     424.150000       2.0

The WindSpeed dataset contains 631149 points while the WindDirection set contains 631144 points.

# Merging WindSpeed and WindDirection datasets 

In [52]:
Ws_Wd_Merged = pd.merge(WindSpeed, WindDirection[['Time','Value']],on = 'Time') # Merging the 2 datasets into one single dataframe called Ws_Wd_Merged.
Ws_Wd_Merged.rename(columns={'Value_x':'Ws_Value', 'Value_y':'Wd_Value'}, inplace= True) # renaming columns
Ws_Wd_Merged.set_index('Time', inplace=True)
Ws_Wd_Merged.describe()
Ws_Wd_Merged.to_csv ('C:\Users\jonat\OneDrive\Documents\Python\Output\Files\Agora\MERGED_wSpeed_wDirection_FINO1_90m_20040101_20151231.csv', index = True, header=True)

Ws_value columns contains the wind speed values while the Wd_value contains wind direction data. No operations have been made on or using any other column data. The merged data set contains 631143. "-999" or "1000" value data points have not been removed. The merged data set is the set of common data points between the raw WindSpeed and WindDirection datasets.
File Name: MERGED_wSpeed_wDirection_FINO1_90m_20040101_20151231.csv

# Identifying missing dates

In [53]:
WindSpeed_Tup = set([tuple(line) for line in WindSpeed['Time'].values])
WindDirection_Tup = set([tuple(line) for line in WindDirection['Time'].values])
WindSpeed_Tup.difference(WindDirection_Tup)
diff_export = pd.DataFrame(list(WindSpeed_Tup.difference(WindDirection_Tup)))
print diff_export
diff_export.to_csv('C:\Users\jonat\OneDrive\Documents\Python\Output\Files\Agora\Missing_Days_wSpeed_wDirection_FINO1_90m_20040101_20151231.csv', index = None, header=None)

  0  1  2  3  4  5  6  7  8  9  10 11 12 13 14 15 16 17 18
0  2  0  0  7  -  1  2  -  0  1     0  0  :  2  0  :  0  0
1  2  0  0  7  -  1  1  -  0  1     0  0  :  1  0  :  0  0
2  2  0  0  7  -  1  2  -  0  1     0  0  :  1  0  :  0  0
3  2  0  0  7  -  1  1  -  0  1     0  0  :  2  0  :  0  0
4  2  0  0  7  -  1  2  -  0  1     0  0  :  3  0  :  0  0
5  2  0  0  7  -  1  1  -  0  1     0  0  :  3  0  :  0  0


The merged data setcontains 631143 points implying that there is a discrepancy of 6 data. Running the difference code outputs the above 6 dates. File Name : Missing_Days_wSpeed_wDirection_FINO1_90m_20040101_20151231.csv